Source/Core/Operations/AsyncOperation.swift

```diff
@@ -16,110 +16,97 @@
 
 import Foundation
 
-internal class AsyncOperation<Result, Cancellable>: Operation, CompletableInSpecificQueue {
+/// Base class for asynchronous operations that will be put in `OperationQueue`
+public class AsyncOperation: Operation, CompletableInSpecificQueue {
     
-    private var completionQueue: DispatchQueue?
-    private var cancellable: Cancellable?
-    private var resultReported = false
+    override public var isAsynchronous: Bool { return true }
+    override public var isReady: Bool { return state == .isReady && dependencies.allSatisfy({ $0.isFinished }) }
```

Comment:
I cannot compile `allSatisfy` on older xcode 9.4.1. We should temporarily reimplement that with our own code, until we switch to newer swift & xcode.